src/scripts/shop/ShopHandler.ts

```diff
@@ -44,7 +44,7 @@ class ShopHandler {
             return input.val(0).change();
         }
         let amt = 1;
-        for (amt; App.game.wallet.hasAmount(new Amount(item.totalPrice(amt), item.currency)); amt++){}
+        for (amt; App.game.wallet.hasAmount(new Amount(item.totalPrice(amt), item.currency)) && amt <= item.maxAmount && amt <= 1e6; amt++){}
```

Comment:
Why `1e6`?

Reply:
1m iterations seems to take around ~500ms to complete, any longer feels too laggy

Reply:
Open the shop and select the Pokeballs _(You need to have enough money)_.
```javascript
// gives enough money
App.game.wallet.currencies[GameConstants.Currency.money] = Number.MAX_SAFE_INTEGER;
// Need to have shop open and item selected (preferably Pokeball);
item  = ShopHandler.shopObservable().items()[ShopHandler.selected()];
// Log time taken
console.time('test')
for (amt = 1; App.game.wallet.hasAmount(new Amount(item.totalPrice(amt), item.currency)) && amt <= 1e6; amt++){}
console.timeEnd('test');
--amt;
```